knowledge_repo/app/auth_providers/oauth2.py

```diff
@@ -112,9 +113,13 @@ def init(self,
         if self.verify_ssl_certs is None:
             self.verify_ssl_certs = True
 
-        redirect_url = self.app.config['SERVER_NAME'] or 'localhost:7000'
+        host = self.app.config['SERVER_NAME'] or 'localhost:7000'
+        scheme = self.app.config['PREFERRED_URL_SCHEME'] or 'http'
+        redirect_url = '{}://{}'.format(scheme, host)
+        redirect_path = '/auth/login/{}/authorize'.format(self.name)
         if self.app.config['APPLICATION_ROOT']:
-            redirect_url = '/'.join((redirect_url, self.app.config['APPLICATION_ROOT']))
+            redirect_path = path.join(self.app.config['APPLICATION_ROOT'], redirect_path)
```

Comment:
Using `path.join` will break this on Windows machines (since the default path separator the is '\'). Consider using `'\'.join()` or if you need `path` for some reason, `posixpath`.

Reply:
Thank you, I completely forgot about Windows :sweat: 
`join` is needed, as the default `APPLICATION_ROOT` is `/`, which would create an invalid path.